File: 240409/숫자 차례로 출력하기/to-output-numerically.py
N = int(input())

def print_number(N):
    if N == 0:
        return

    print_number(N - 1)
    print(N, end = ' ')

def print_rev_number(N):
    if N == 0:
        return

    print(N, end = ' ')
    print_rev_number(N - 1)

print_number(N)
print()
print_rev_number(N)

File: 251224/숫자 차례로 출력하기/to-output-numerically.cpp
#include <iostream>

using namespace std;

int N;
void print_number(int num);
void print_rev_number(int num);

int main() 
{
    cin >> N;

    // Please write your code here.
    print_number(N);
    cout << endl;
    print_rev_number(N);
    return 0;
}

void print_number(int num)
{
    if(num == 0)
        return ;

    print_number(num - 1);
    cout << num << ' ';
}

void print_rev_number(int num)
{
    if(num == 0)
        return ;

    cout << num << ' ';
    print_rev_number(num - 1);
}